Userland/Games/Solitaire/Game.cpp

```diff
@@ -482,6 +498,76 @@ void Game::paint_event(GUI::PaintEvent& event)
     }
 }
 
+void Game::undo_save_move(CardStack& from, CardStack& to, NonnullRefPtrVector<Card> moved_cards)
+{
+    m_last_move.type = LastMove::Type::MoveCards;
+    m_last_move.from = &from;
+    m_last_move.cards = moved_cards;
+    m_last_move.to = &to;
+    if (on_undo_available)
+        on_undo_available(true);
+}
+
+void Game::undo_save_flip(Card& card)
+{
+    NonnullRefPtrVector<Card> cards;
+    cards.append(card);
+    m_last_move.type = LastMove::Type::FlipCard;
+    m_last_move.cards = cards;
+    if (on_undo_available)
+        on_undo_available(true);
+}
+
+void Game::perform_undo()
+{
+    if (!m_last_move.from)
+        return;
```

Comment:
Won't this prevent undoing a flip? `undo_save_flip` doesn't set `m_last_move.from`.

Reply:
Yeah I believe so, good catch.